src/demoserver.cpp

```diff
@@ -0,0 +1,294 @@
+#include "demoserver.h"
+#include "lobby.h"
+
+DemoServer::DemoServer(QObject *parent) : QObject(parent)
+{
+    timer = new QTimer(this);
+    timer->setTimerType(Qt::PreciseTimer);
+    timer->setSingleShot(true);
+
+    tcp_server = new QTcpServer(this);
+    connect(tcp_server, &QTcpServer::newConnection, this, &DemoServer::accept_connection);
+    connect(timer, &QTimer::timeout, this, &DemoServer::playback);
+}
+
+void DemoServer::start_server()
+{
+    if (server_started) return;
+    if (!tcp_server->listen(QHostAddress::Any, 0)) {
```

Comment:
Don't listen on all interfaces or it will trigger the Windows firewall.

```suggestion
    if (!tcp_server->listen(QHostAddress::LocalHost, 0)) {
```

Reply:
this stops it from working

Reply:
i am retard